src/hotspot/share/oops/compressedKlass.hpp

```diff
@@ -56,9 +56,23 @@ class CompressedKlassPointers : public AllStatic {
   //  could use this info to optimize encoding.
   static size_t _range;
 
-  static void set_base(address base);
+  // Reduce the number of loads generated to decode an nKlass by packing all
+  // relevant information (flag, base, shift) into a single 64-bit word. The
+  // encoding base is aligned to metaspace reserve alignment (16Mb), so enough
+  // space to hide additional info away in low bytes. It even allows us the luxury
+  // to use 8 bits for shift and flag each, resulting in 8-bit moves used by the
+  // compiler.
+  // - Bit  [0-7]   shift
+  // - Bit  8       UseCompressedClassPointers
+  // - Bits [16-64] the base.
+  static uint64_t _combo;
```

Comment:
```suggestion
  static uint64_t _compressionInfo;
```

Otherwise, IMO this patch is good to go.